### ISSUE TITLE
An Explaination on How Combat Works?

### ISSUE THREAD

**Issue by Xebenkeck** [Sh0wdown/UnofficialCrusaderPatch#483]
Hello,

I was just wondering if there is a guide somewhere as to how the combat/unit balance works in Crusader.

In the official patch some unit changes happened, like the Arab swordsman crossbow defense.

Are these stats in the code that can be adjusted? If so is there a list of them?

Cause I imagine there are stats like:
Arrow Defense
Bolt Defense
Rock Defense
Fire Defense
Wall Arrow Defense?
Wall Arrow Attack?
Melee Attack
Melee Defense
Melee Unit Multiplicative Stat(like how 3 Monks fighting together are more powerful 3 Mace-men fighting together but 1 v 1 a Mace-man wins)
+ Fear Factor
- Fear Factor
Charge Bonus?
Arrow Damage
Bolt Damage
Rock Damage
ETC.

I'm just curious because for unit balance, even to make suggestions it would helpful if we knew what stats/attributes can be adjusted.

As of the current patch, I'd say there are two pretty big unit balance issues. 

First, Monks die just as fast as Spear-men to arrows, bolts. I feel they should have the Same Arrow/Bolt defense as Mace-men. Yes Mace-men require two weapons to make, where monks don't, but monks are less powerful attack-wise, individually, can't climb ladders, can't dig moats and are way slower, so I think they'd be more balanced that way. The Abbot feels really under-powered attack-wise. Just run a Abbot vs Snake/PIG/Wolf and just laugh at the monks dying instantly to archers. The only thing that makes him even remotely potent is the catapults he builds.

Secondly, Slingers are woefuly underpowered. They have less range then all other ranged units, yet when they do get into range the payoff for that is none existent. I feel their own defense is fine they are cheap after all, but their attack power against melee units specifically could be significantly buffed as a reward for getting them into range to attack.

**Comment by GRhin** [comment]
This is a duplicate #478

**Comment by GRhin** [comment]
As for the two underpowered items you mention, fully disagree. The two units are both anti-melee units, and are seriously powerful when used for that purpose, if you have been following krarilotus's Christmas campaign there are several times when you have to use those units as your primary defense to survive, no other unit can compare to them in their strength. 
In multiplayer, when playing vanilla rules on vanilla maps, I often add slingers to my assault force mixed with ha and shields to kill any knights or assassins that try to push me off, their lesser range is a benefit in this case cos ranged units by default target ranged units first, so your horse archers are busy shooting enemy ha behind shields and ignoring the melee while generally the ha are out of range of my slings so they are free to target the melee first. And they are pretty much as good as crossbows in dps due to their reload speed, better in some cases.

**Comment by Mizo5119** [comment]
Yeah Slingers are really good for early game assaults and reinforcements.

Monks are fine too. They're very spammy.

**Comment by J-T-de** [comment]
Regarding what parameters exist, https://github.com/Sh0wdown/UnofficialCrusaderPatch/issues/271 shows some parameters.

**Comment by Xebenkeck** [comment]
Basically what I'm getting is all the unit stats aren't known yet/explored as they still coded in hex?

Also I am basing the balnce of those units off of AI. Humans no doubt can use them far better than CPU's can. But any match I have with the Sultan or Abbot in them they get destroyed if the game goes past 10 mins by virtually all the other AI lords.

I myself have my own AIC and AIV's going and have tampered with their balances, while still keeping their "character" in tact. And with improved defenses for most AI's, Monks in particular are very underwhelming to watch try siegeing a keep. If i wish to keep the Abbot Archer and Monk, then most I would say he is the third weakest lord. Ya his economy is second to none, but his troops are abysmal. 

But if Sh0Wdown makes an upcoming patch with changeable parameters, then I guess we all can change it to our desires.

**Comment by GRhin** [comment]
That's like saying slaves need improvement because snake dies. What you need to do is modify the ais to use the units effectively, instead of overpowering them so that the ai doesn't need fixing. Check out krarilotus's aggressive ai for a implementation of this methodology. All ais stay true to their core personalities, but attack earlier, harder and faster. Abbot get some swords to help and I think so seige to help kill the ranged so the monks are more effective, and abbot gets more archers (which he already used)
